cmds/core/ip/route_linux_test.go

```diff
@@ -62,211 +59,287 @@ func TestParseRouteAddAppendReplaceDel(t *testing.T) {
 		return 0, fmt.Errorf("test error: link %s not found", name)
 	}
 
-	tests := []struct {
-		name     string
-		args     []string
-		expected netlink.Route
-		wantErr  bool
+	// nlRoute is a custom route type to shadow netlink.Route.
+	type nlRoute netlink.Route
+
+	// ignoreRouteMethods is a transformer to be used with cmp.Diff to transform
+	// netlink.Route to nlRoute, which is a type alias of netlink.Route, but
+	// does not implement any methods. Especially it does not implement the
+	// Equal() method, which is used by cmp.Diff to compare the two routes and
+	// the String() method, which is used by cmp.Diff to print the diff.
+	ignoreRouteMethods := cmp.Transformer("", func(r netlink.Route) nlRoute {
+		return nlRoute(r)
+	})
+
+	// nlNexthopInfo is a custom type to shadow netlink.NexthopInfo.
+	type nlNexthopInfo netlink.NexthopInfo
+
+	// ignoreNexthopInfoMethods is a transformer to be used with cmp.Diff to
+	// transform netlink.NexthopInfo to nlNexthopInfo, which is a type alias of
+	// netlink.NexthopInfo, but does not implement any methods. Especially it
+	// does not implement the Equal() method, which is used by cmp.Diff to
+	// compare the two nexthops and the String() method, which is used by
+	// cmp.Diff to print the diff.
+	ignoreNexthopInfoMethods := cmp.Transformer("", func(nh netlink.NexthopInfo) nlNexthopInfo {
+		return nlNexthopInfo(nh)
+	})
+
+	validTests := []struct {
+		name string
+		args []string
+		want *netlink.Route
 	}{
-		{
-			name:    "fail to parse dst",
-			args:    []string{"abc"},
-			wantErr: true,
-		},
-		{
-			name: "Add route with valid arguments",
-			args: []string{"192.0.0.2/24", "dev", "lo"},
-			expected: netlink.Route{
-				LinkIndex: linkToIdx["lo"],
-				Dst:       dst,
+		// {
+		// 	name: "Add route with prefix and device",
+		// 	args: []string{"192.168.1.0/24", "dev", "eth0"},
+		// 	want: &netlink.Route{
+		// 		Dst: &net.IPNet{IP: net.IPv4(192, 168, 1, 0), Mask: net.CIDRMask(24, 32)},
+		// 		MultiPath: []*netlink.NexthopInfo{
+		// 			{
+		// 				LinkIndex: linkToIdx["eth0"],
+		// 			},
+		// 		},
+		// 	},
+		// },
```

Comment:
These commented test definitions seem to be semantically equivalent to their following cases and seem old-ish, so they can also be removed?

Reply:
removed